src/usr/local/www/services_dhcp.php

```diff
@@ -235,6 +243,76 @@ function validate_partial_mac_list($maclist) {
 
 	/* input validation */
 
+	/*
+	 * Check the OMAPI settings
+	 * - Make sure that if the port is defined, that it is valid and isn't in use
+	 * - Make sure the key is defined and the length is appropriate for the selected algorithm
+	 * - Generate a new key if selected
+	 */
+	if (!empty($_POST['omapi_port'])) {
+		// Check the port entry
+		if (is_port($_POST['omapi_port']) && $_POST['omapi_port'] > 1024) {
+			// Get the list of open or listening ports locally
+			exec("/usr/bin/netstat -an4p tcp | /usr/bin/awk '{print $4}' | /usr/bin/egrep -o '*([0-9]{1,5})$' | /usr/bin/sort -u", $port_info);
```

Comment:
The more I think about it, this would be generally useful in its own function so other areas could use it as well. I'd like to see this check moved to `/etc/inc/util.inc` in a function called `is_port_in_use($port)` that returns true/false based on whether the passed port is already in use. Then you could use that function in the if statement here and then add to input errors based on the result.

Reply:
Will do!